### DISCUSSION TITLE
Dropdown menu PlusToolkitServer

### DISCUSSION THREAD

**Discussion by D49ll** [PlusToolkit/PlusLib#1144] [Q&A] [answered]
Hi,

I have added a new device to PlusToolkit. Currently i've only created the bare minimum in vtkPlusMyDevice.cxx/h and tested the functions in debug mode via vtkPlusMyDeviceTest. I followed the guide provided on https://plustoolkit.github.io/devicecode.

I wanted to try and start a server from the PlusServerLaucher app, however i cannot find my solution in the dropdown menu. 

![image](https://github.com/PlusToolkit/PlusLib/assets/47644255/84fb46e8-23b3-4554-80a6-4b5b71bf8e84)

Where and what do i need to do to add my device to the dropdown menu. Or is it possible that the build process PlusBuild via cmake has not been performed correctly by me?

**Comment by Sunderlandkyl** [comment, marked as answer]
The list is populated from the config files in the config file directory.

**Reply by D49ll** [reply]
Thanks for your reply!

By config file directory you mean this 

C:\dev\PlusB-bin\PlusLibData\ConfigFiles

If so, ive added a "PlusDeviceSet_Server_MyDevice.xml" to this directory. However i still dont get an option when rebuilding.

Is there something else i need to add?

**Reply by Sunderlandkyl** [reply]
Can you attach the config file?

**Reply by D49ll** [reply]
```
<PlusConfiguration version="2.1">

  <DataCollection StartupDelaySec="1.0" >
    <DeviceSet 
      Name="PlusServer:  MNS tracker with passive markers" 
      Description="Broadcasting tool tracking data through OpenIGTLink
For  MNS passive marker starting kit: Tool (1001), Reference (1002)" />
    <Device
      Id="TrackerDevice"
      Type="MNSTracker"
      ToolReferenceFrame="Tracker" >
      <DataSources>
        <DataSource Type="Tool" Id="Tool" UnitId = "1001"/>
        <DataSource Type="Tool" Id="Reference" UnitId = "1002"/>
      </DataSources>
      <OutputChannels>
        <OutputChannel Id="TrackerStream" >
          <DataSource Id="Tool"/>
          <DataSource Id="Reference"/>
        </OutputChannel>
      </OutputChannels>
    </Device>
  </DataCollection>

  <PlusOpenIGTLinkServer 
    MaxNumberOfIgtlMessagesToSend="1"
    MaxTimeSpentWithProcessingMs="50"
    ListeningPort="18944"
    SendValidTransformsOnly="true"
    OutputChannelId="TrackerStream" >
    <DefaultClientInfo>
      <MessageTypes>
        <Message Type="TRANSFORM" />
      </MessageTypes>
      <TransformNames>
        <Transform Name="ToolToReference" />
        <Transform Name="ToolToTracker" />
        <Transform Name="ReferenceToTracker" />
      </TransformNames>
    </DefaultClientInfo>
  </PlusOpenIGTLinkServer>

</PlusConfiguration>

```

**Reply by D49ll** [reply]
Could the issue be due to a name error in the "adding a device"-setup? I cant seem to find any instructions on what to add in the ConfigFiles directory

**Reply by Sunderlandkyl** [reply]
Not sure, can you attach a log file as well?

**Reply by D49ll** [reply]
I got it to work. I changed the filename from MNS to MNSTracker (which is what i call my device).

![image](https://github.com/PlusToolkit/PlusLib/assets/47644255/70705222-9f97-43bf-a617-f1a2ab7908a1)
